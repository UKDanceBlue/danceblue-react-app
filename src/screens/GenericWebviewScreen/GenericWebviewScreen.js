// Import third-party dependencies
import React from 'react';
import { View } from 'react-native';
import { WebView } from 'react-native-webview';
import Constants from 'expo-constants';

/**
 * A simple screen wrapper for a webview
 * @param {Object} route The route used by the navigator to reach this screen, should include a defaultParam *uri* that will be displayed by the component
 */
const GenericWebviewScreen = ({ route }) => {
  let { uri } = route.params;
  if (!uri && uri != '') {
    // If a uri is not given (and I don't mean blank) then send a 404 from DanceBlue's website
    uri = 'https://www.danceblue.org/404/';
  }
  return (
    <View style={{ flex: 1 }}>
      <WebView source={{ uri }} />
    </View>
  );
};

export default GenericWebviewScreen;
