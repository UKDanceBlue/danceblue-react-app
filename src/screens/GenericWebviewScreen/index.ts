import GenericWebviewScreen from "./GenericWebviewScreen";

export default GenericWebviewScreen;
