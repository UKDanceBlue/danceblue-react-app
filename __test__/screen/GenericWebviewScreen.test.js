import React from 'react';
import renderer from 'react-test-renderer';
import GenericWebviewScreen from '../../src/screens/GenericWebviewScreen';

test('renders correctly', () => {
  const tree = renderer
    .create(<GenericWebviewScreen route={{ params: { uri: 'www.google.com' } }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
