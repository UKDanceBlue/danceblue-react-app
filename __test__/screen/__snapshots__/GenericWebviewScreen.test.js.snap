// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic webview renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadEnd={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "www.example.com",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;

exports[`generic webview shows 404 with no uri 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadEnd={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "https://www.danceblue.org/404/",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
